packages/hooks/src/useIsRtl.ts

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+const useIsRtl = () => {
+    const { i18n } = useTranslation();
+
+    const check_rtl = React.useCallback(() => {
+        return i18n.dir(i18n.language?.toLowerCase()) === 'rtl';
+    }, [i18n]);
+
+    const [is_rtl, setIsRtl] = React.useState<boolean>(() => {
+        return check_rtl();
+    });
```

Comment:
```suggestion
    const [is_rtl, setIsRtl] = React.useState(check_rtl());
```
should infer the return type of check_rtl as boolean automatically. :)

Reply:
`    const [is_rtl, setIsRtl] = React.useState(check_rtl());`
putting the function like this in useState will call the `check_rtl` function to be called on each render, if the use-case if to check for `rtl` on just mount then the implemented way is correct, it just can be cleaned 
```suggestion
    const [is_rtl, setIsRtl] = React.useState<boolean>(() => check_rtl());
```